queue_1.py

```diff
@@ -0,0 +1,64 @@
+from list import List
+
+
+class QueueIterator:
+    def __init__(self, head) -> None:
+        self.current = head
+
+    
+    def __iter__(self):
+        return self
+
+
+    def __next__(self):
+        if self.current == None:
+            raise StopIteration
+
+        elem = self.current.elem
+        self.current = self.current.next
+        return elem
+
+
+class Queue:
+    def __init__(self):
+        self.list = List()   
+
+
+    def __len__(self):
+        return self.list.length()
+
+
+    def length(self):
+        return self.list.length()
+
+
+    def __eq__(self, other: 'List'):
+        return self.list.__eq__(other)
+
+
+    def __getitem__(self, index):
+        return self.list.__getitem__(index)
+
+
+    def __setitem__(self, index, elem):
+        self.list.__setitem__(index, elem)
+
+
+    def __str__(self) -> str:
+        return str(self.list)
+
+
+    def __iter__(self):
+        return QueueIterator(self.list.head)
```

Comment:
Let's try to do it this way. Let's make *all* fieds of `class List` private. See [this article](https://www.tutorialsteacher.com/python/public-private-protected-modifiers) for example. It talks about public, protected, and private members (where the term `member` is a synonym to `field or method`). The "protected" privacy is actually of no interest for us right now (and in fact it doesn't even exist in `Rust`, for example), so let's skip it.

From now on you should explicitly identify which members (fields and methods) are private and which are public by adding `__` suffix to the names of private members.

For example the `List` class should have the fields named `__head`, `__length` and `__tail`. It's rather ugly syntax, I agree, but this is how Python implements privacy, and how this language has been working for decades, so let's follow this approach in Python.

Reply:
[Here](9edb0faf05b0e47b5fa7febaaac077c16ff60368) I have defined and done:
- all the fields (`head`, `tail`, `length`) of the list are now private,
- `find_node` and `check_index` are now private. Since these are internal methods of the `list` class and they should not be included in the user interface,
- the `Node` and `ListIterator` classes will not be imported via `from list import *`; but they will still be available if I specify them myself: `from list import _Node`. I found this information [here](https://stackoverflow.com/questions/551038/private-implementation-class-in-python).

I think that the last point is inherent in Python. In other languages it will be easier with this.
But it seems to me that I have understood a little the meaning of using public and private modifiers 😅

Reply:
Good, the understanding that files (modules), classes and functions are just like user interfaces is the right one